File: inject/sample_without_static_field/app/controllers/Application.java
package controllers;

import play.*;
import play.mvc.*;

import views.html.*;

import javax.inject.*;
import service.*;

public class Application extends Controller {

  private MyService s;

  @Inject public Application( MyService s) {
    this.s=s;
  }
  
  public  Result index() {
    return ok(index.render(s.demonstrate()));
  }
  
}

File: statsd/sample/sample-statsd/app/controllers/Application.scala
package controllers

import play.api.mvc._
import play.api.libs.concurrent.Execution.Implicits._
import scala.concurrent.Future

object Application extends Controller {
  def index = Action {
    Thread.sleep(2)
    Ok
  }

  def singleParam(p: String) = Action {
    Thread.sleep(2)
    Ok
  }

  def twoParams(p1: String, p2: String) = Action {
    Thread.sleep(2)
    Ok
  }

  def async = Action.async {
    Future {
      Thread.sleep(2)
      Ok
    }
  }

  def syncFailure = Action {
    Thread.sleep(2)
    if (true) throw new RuntimeException
    Ok
  }

  def asyncFailure = Action.async {
    Future {
      Thread.sleep(2)
      if (true) throw new RuntimeException
      Ok
    }
  }

  def error = Action {
    Thread.sleep(2)
    ServiceUnavailable
  }
}
